hypercall.h

```diff
@@ -1,168 +1,118 @@
-#ifndef MAGIC_VALUE
-#error Needs MAGIC_VALUE
-#endif
-#define RETRY 0xDEADBEEF
-// #if defined(__x86_64__)
-#if defined(__x86_64__)
-static inline int hc(int hc_type, void **s,int len) {
-    uint64_t eax = MAGIC_VALUE;
-    uint64_t ret = hc_type;
-    int y = 0;
-    do {
-        ret = MAGIC_VALUE;
-        volatile int x = 0;
-        for(int i = 0; i< len; i++) {
-            x |= *(((int*)s[i])+y);
-        }
-        asm __volatile__(
-        "movq %1, %%rax \t\n\
-        movq %2, %%rdi \t\n\
-        movq %3, %%rsi \t\n\
-        movq %4, %%rdx \t\n\
-        cpuid \t\n\
-        mov %%rax, %0 \t\n"
-        : "=m" (ret) /* output operand */
-        : "g" (eax), "g" ((uint64_t)hc_type), "g" (s), "g" ((uint64_t)len) /* input operands */
-        : "rdi", "rsi", "rdx", "eax", "ebx", "ecx", "edx" /* clobbered registers */
```

Comment:
Don't we want to say some registers can be clobbered by a hypercall - not sure if this should've been deleted.

Reply:
Am I missing that somewhere? I just see the `memory` keyword, e.g., https://github.com/panda-re/libhc/pull/7/files#diff-29186ed653452f4fa83e7a65ed4853b76bb360174f20b93255c5e4d4b9d0fcd2R73

Reply:
I could be wrong about register keyword not clobbering in this case but I would've expected errors before now.
https://github.com/panda-re/libhc/pull/7/files#diff-29186ed653452f4fa83e7a65ed4853b76bb360174f20b93255c5e4d4b9d0fcd2R2

Reply:
Ah yeah, okay - I think you're right. False alarm sorry